sklearn/tests/test_common.py

```diff
@@ -204,3 +207,64 @@ def test_class_support_removed():
 
     with pytest.raises(TypeError, match=msg):
         parametrize_with_checks([LogisticRegression])
+
+
+class MyNMFWithBadErrorMessage(NMF):
+    # Same as NMF but raises an uninformative error message if X has negative
+    # value. This estimator would fail the check suite in strict mode,
+    # specifically it would fail check_fit_non_negative
+    def fit(self, X, y=None, **params):
+        X = check_array(X, accept_sparse=('csr', 'csc'),
+                        dtype=[np.float64, np.float32])
+        try:
+            check_non_negative(X, whom='')
+        except ValueError:
+            raise ValueError("Some non-informative error msg")
+
+        return super().fit(X, y, **params)
+
+
+def test_strict_mode_check_estimator():
```

Comment:
Just curious how long does this test takes? Running `check_estimator` on an estimator takes a while and we should likely avoid it if possible, though here I guess we have no choice?

Reply:
about 6 seconds on my machine. I agree it's a bit long but I think it's worth it?